unstructured/embed/interfaces.py

```diff
@@ -0,0 +1,26 @@
+from abc import ABC, abstractmethod
+from typing import List, Optional
+
+from unstructured.documents.elements import Element
+
+
+class BaseEmbeddingEncoder(ABC):
+    @abstractmethod
+    def initialize():
+        """Initializes the embedding encoder class. Should also validate the instance
+        is properly configured: e.g., embed a single a element"""
+        pass
+
+    @property
+    def num_of_dimensions(self) -> int:
```

Comment:
Should this method be an abstract method and require the implementation of this base class to define it? Otherwise the return type should be `Optional[int]` since the default behavior is to return `None`.

Reply:
Agreed, and addressed with https://github.com/Unstructured-IO/unstructured/pull/1368/commits/04468f0e5f6e19546dfa4ef17ef8d4a2a7064f71

Reply:
Please also note https://github.com/Unstructured-IO/unstructured/pull/1368/commits/8c51d8ade6423cda7085cb2a3f6f0927dd2d1878